server/src/main/java/com/flipkart/varadhi/utils/AsyncProducerReadStream.java

```diff
@@ -0,0 +1,90 @@
+package com.flipkart.varadhi.utils;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.streams.ReadStream;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AsyncProducerReadStream implements ReadStream<Buffer> {
```

Comment:
is there a resource which you followed for this?

Reply:
chat gpt.